File: Python/13RomantoInteger.py
__author__ = 'minyi'
'''
Given a roman numeral, convert it to an integer.
Input is guaranteed to be within the range from 1 to 3999.
'''

class Solution(object):
    def romanToInt(self, s):
        """
        :type s: str
        :rtype: int
        """
        dict = {
            'I':1,
            'V':5,
            'X':10,
            'L':50,
            'C':100,
            'D':500,
            'M':1000
        }

        ans = dict[s[0]]

        for i in range(1,len(s)):
            if dict[s[i-1]]<dict[s[i]]:
                ans += dict[s[i]] - 2*dict[s[i-1]]
            else:
                ans += dict[s[i]]

        return ans

s = Solution()
print(s.romanToInt())

File: CPP/13RomantoInteger.cpp
#include <iostream>
#include <cstring>
#include <map>

using namespace std;

class Solution {
public:
    int romanToInt(string s) {
    	int ans = toNum(s[0]);
    	for(int i=1;i<s.size(); i++){
    		if(toNum(s[i-1])<toNum(s[i])){
    			ans += toNum(s[i]) - 2*toNum(s[i-1]);
			}else{
				ans += toNum(s[i]);
			}
		}
		return ans;
    }
    
    int toNum(char c){
    	switch(c){
    		case 'I':return 1;
    		case 'V':return 5;
    		case 'X':return 10;
    		case 'L':return 50;
    		case 'C':return 100;
    		case 'D':return 500;
    		case 'M':return 1000;
    	}
	}
};

int main(){
	return 0;
}
